### PR TITLE
MONGOID-4506 Support sessions

### PR DESCRIPTION
**Pull request by estolfo**

### CODE REVIEW

File: spec/mongoid/clients/sessions_spec.rb [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+        before do
+          Person.with_session do
+            Person.create
+            Person.create
+          end
+        end
+
+        let(:insert_events) do
+          subscriber.started_events.select { |event| event.command['insert'] }
+        end
```
Comment (ShaneHarvey) [inline spec/mongoid/clients/sessions_spec.rb]: Is this a duplicate of the above `:insert_events`?

File: spec/mongoid/clients/sessions_spec.rb [hunk 2]
```diff
@@ -0,0 +1,297 @@
+require "spec_helper"
```
Comment (ShaneHarvey) [inline spec/mongoid/clients/sessions_spec.rb]: Should there be any tests that use multiple threads?

File: spec/mongoid/clients/sessions_spec.rb [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+        end
+
+        let(:insert_events) do
+          subscriber.started_events.select { |event| event.command['insert'] }
+        end
+
+        it 'uses a single session id for all operations on the class' do
+          expect(Person.count).to be(2)
+          expect(insert_events).not_to be_empty
+          expect(insert_events.collect { |event| event.command['lsid'] }.uniq.size).to eq(1)
```
Comment (ShaneHarvey) [inline spec/mongoid/clients/sessions_spec.rb]: This doesn't actually test that every command sent a session id. I suggest:
```
lsids_sent = insert_events.collect { |event| event.command['lsid'] }
expect(lsids_sent.size).to eq(2)
expect(lsids_sent.uniq.size).to eq(1)
 ```

Same for the other tests. I also suggest adding a test that runs every operation that a Mongoid class can do to make sure we didn't miss adding a session in some place. Please ignore if that's already the case.

File: spec/mongoid/clients/sessions_spec.rb (hunk 1 above)
Comment (estolfo) [reply spec/mongoid/clients/sessions_spec.rb]: Yes, indeed. I'll remove this, thanks.

File: spec/mongoid/clients/sessions_spec.rb (hunk 3 above)
Comment (estolfo) [reply spec/mongoid/clients/sessions_spec.rb]: OK, good point. I'll change this.

File: spec/mongoid/clients/sessions_spec.rb (hunk 2 above)
Comment (estolfo) [reply spec/mongoid/clients/sessions_spec.rb]: Sure, that's a good idea. I'll add a test to make sure a session is not accessible from another thread.

**Pull request merged**